Makefile.am

```diff
@@ -43,6 +43,7 @@ common_srcs =				\
 	src/hmem.c			\
 	src/hmem_rocr.c			\
 	src/hmem_cuda.c			\
+	src/hmem_cuda_gdrcopy.c		\
```

Comment:
Can we just fold this into hmem_cuda.c and skip adding a new source file?

Also, please update the commit message to describe why we want gdrcopy instead of cuda memcpy.

Reply:
We can fold the file into hmem_cuda.c, but it would make the resulted file not easy to read IMO. I would prefer to introduce a new file.

Updated the commit message to add the reasoning behind this PR.